fuzzing/valuegenerationtracer/valuegeneration_tracer.go

```diff
@@ -0,0 +1,357 @@
+}
+
+// resolveConstructorArgs resolves previously unresolved constructor argument ABI data from the call data, if
+// the call frame provided represents a contract deployment.
+func (v *ValueGenerationTracer) resolveCallFrameConstructorArgs(callFrame *utils.CallFrame, contract *contracts.Contract) {
+	// If this is a contract creation and the constructor ABI argument data has not yet been resolved, do so now.
+	if callFrame.ConstructorArgsData == nil && callFrame.IsContractCreation() {
+		// We simply slice the compiled bytecode leading the input data off, and we are left with the constructor
+		// arguments ABI data.
+		compiledInitBytecode := contract.CompiledContract().InitBytecode
+		if len(compiledInitBytecode) <= len(callFrame.InputData) {
+			callFrame.ConstructorArgsData = callFrame.InputData[len(compiledInitBytecode):]
+		}
+	}
+}
+
+// resolveCallFrameContractDefinitions resolves previously unresolved contract definitions for the To and Code addresses
+// used within the provided call frame.
+func (v *ValueGenerationTracer) resolveCallFrameContractDefinitions(callFrame *utils.CallFrame) {
+	// Try to resolve contract definitions for "to" address
+	if callFrame.ToContractAbi == nil {
+		// Try to resolve definitions from compiled contracts
+		toContract := v.contractDefinitions.MatchBytecode(callFrame.ToInitBytecode, callFrame.ToRuntimeBytecode)
+		if toContract != nil {
+			callFrame.ToContractName = toContract.Name()
+			callFrame.ToContractAbi = &toContract.CompiledContract().Abi
+			v.resolveCallFrameConstructorArgs(callFrame, toContract)
+
+			// If this is a contract creation, set the code address to the address of the contract we just deployed.
+			if callFrame.IsContractCreation() {
+				callFrame.CodeContractName = toContract.Name()
+				callFrame.CodeContractAbi = &toContract.CompiledContract().Abi
+			}
+		}
+	}
+
+	// Try to resolve contract definitions for "code" address
+	if callFrame.CodeContractAbi == nil {
+		codeContract := v.contractDefinitions.MatchBytecode(nil, callFrame.CodeRuntimeBytecode)
+		if codeContract != nil {
+			callFrame.CodeContractName = codeContract.Name()
+			callFrame.CodeContractAbi = &codeContract.CompiledContract().Abi
+		}
+
+	}
+}
+
+// getCallFrameReturnValue generates a list of elements describing the return value of the call frame
+func (t *ValueGenerationTracer) getCallFrameReturnValue() TransactionOutputValues {
+	// Define some strings that represent our current call frame
+	var method *abi.Method
+
+	// Define a slice of any to represent return values of the current call frame
+	var outputValues TransactionOutputValues
+
+	// Resolve our method definition
+	if t.currentCallFrame.CodeContractAbi != nil {
+		method, _ = t.currentCallFrame.CodeContractAbi.MethodById(t.currentCallFrame.InputData)
+	}
+
+	if method != nil {
+		// Unpack our output values and obtain a string to represent them, only if we didn't encounter an error.
+		if t.currentCallFrame.ReturnError == nil {
```

Comment:
redundant check - you can remove the check to ensure that an error has not been returned